src/snowflake/snowpark/modin/plugin/extensions/base_overrides.py

```diff
@@ -6,31 +6,152 @@
 should be done with discretion, and only when relevant changes cannot be made to the query compiler or
 upstream frontend to accommodate Snowpark pandas.
+
+If you must override a method in this file, please add a comment describing why it must be overridden,
+and if possible, whether this can be reconciled with upstream Modin.
 """
 from __future__ import annotations
 
 import pickle as pkl
-from typing import Any
+import warnings
+from collections.abc import Sequence
+from typing import Any, Callable, Hashable, Literal, Mapping, cast, get_args
 
+import modin.pandas as pd
 import numpy as np
+import numpy.typing as npt
 import pandas
 from modin.pandas.base import BasePandasDataset
-from pandas._libs.lib import no_default
+from pandas._libs import lib
+from pandas._libs.lib import NoDefault, is_bool, no_default
 from pandas._typing import (
+    AggFuncType,
+    AnyArrayLike,
+    Axes,
     Axis,
     CompressionOptions,
+    FillnaOptions,
+    IgnoreRaise,
+    IndexKeyFunc,
+    IndexLabel,
+    Level,
+    NaPosition,
+    RandomState,
+    Scalar,
     StorageOptions,
     TimedeltaConvertibleTypes,
+    TimestampConvertibleTypes,
+)
+from pandas.core.common import apply_if_callable
+from pandas.core.dtypes.common import (
+    is_dict_like,
+    is_dtype_equal,
+    is_list_like,
+    is_numeric_dtype,
+    pandas_dtype,
+)
+from pandas.core.dtypes.inference import is_integer
+from pandas.core.methods.describe import _refine_percentiles
+from pandas.errors import SpecificationError
+from pandas.util._validators import (
+    validate_ascending,
+    validate_bool_kwarg,
+    validate_percentile,
 )
 
+import snowflake.snowpark.modin.pandas as spd
 from snowflake.snowpark.modin.pandas.api.extensions import (
     register_dataframe_accessor,
     register_series_accessor,
 )
+from snowflake.snowpark.modin.pandas.utils import (
+    ensure_index,
+    extract_validate_and_try_convert_named_aggs_from_kwargs,
+    get_as_shape_compatible_dataframe_or_series,
+    is_scalar,
+    raise_if_native_pandas_objects,
+    validate_and_try_convert_agg_func_arg_func_to_str,
+)
 from snowflake.snowpark.modin.plugin._internal.telemetry import (
+    TELEMETRY_PRIVATE_METHODS,
+    PropertyMethodType,
     snowpark_pandas_telemetry_method_decorator,
 )
-from snowflake.snowpark.modin.plugin.utils.error_message import base_not_implemented
+from snowflake.snowpark.modin.plugin._typing import ListLike
+from snowflake.snowpark.modin.plugin.utils.error_message import (
+    ErrorMessage,
+    base_not_implemented,
+)
+from snowflake.snowpark.modin.plugin.utils.warning_message import WarningMessage
+from snowflake.snowpark.modin.utils import validate_int_kwarg
+
+
+def register_base_override(method_name: str):
+    def decorator(base_method: Any):
+        if callable(base_method) and (
+            not method_name.startswith("_")
+            or (method_name in TELEMETRY_PRIVATE_METHODS)
```

Comment:
also, now since the base class is not declared with metaclass=TelemetryMeta, does that mean we will miss telemetry the comes from the base class? if that is the case, this could be a regression for telemetry cc @sfc-gh-azhan

Reply:
@sfc-gh-joshi Let's add a test to make sure we have telemetry for those cases. E.g., `series.mean` which does not have a series mean method but only use the base's mean method.

Reply:
Good point. `mean` is actually overridden here, so it gets our telemetry annotation, but I've added a test for `copy`, which is inherited straight from `BasePandasDataset`. I had to add a little extra logic to `TelemetryMeta` to make it work, and I'll revisit it when I remove dataframe/series.py.

Reply:
`iloc` is overridden within this file, and it already has an existing test in `test_telemetry`.

Reply:
nice!

Reply:
@sfc-gh-joshi Does that mean for a method that is just defined in base, if they are not overridden here with the register_base_override annotation, we will not have the telemetry. Till the end,  we will align most implementation in this file with modin, and do not have much override here, what happens then?

Reply:
> if they are not overridden here with the register_base_override annotation, we will not have the telemetry.

Yes, though for now, I've pushed a change to `TelemetryMeta` to force Series/DataFrame to apply the telemetry annotation to methods they inherit from BasePandasDataset.

> Till the end, we will align most implementation in this file with modin, and do not have much override here, what happens then?

As I mentioned above: one possible approach would be to use the plugin system to iterate over all of Series/DF's methods, and replace them with a telemetry-annotated version.